### PR TITLE
BuildState with message

### PR DESCRIPTION
**Pull request by BlythMeister**
### Description

* Update BuildState to include a string option for a message
* Update all listeners to work with new TraceData

This should not be considered a breaking change since the code itself states that changing `TraceData` is not considered a breaking change.

> /// Note: Adding new cases to this type is not considered a breaking change!
> /// Please consider not using a match on this type in code external to the fake repository.

Extends #2138 to prevent later conflicts on the TeamCity code so includes all commits from that PR.

### CODE REVIEW

**Comment by matthid** [comment]
The build is red an I noticed 
image

apparently we shouldn't trace processes used for ci messages...

**Comment by matthid** [comment]
Can you please merge latest (build should be green again)?
I guess I should open my own PRs such that release/next is always green :/

**Comment by BlythMeister** [comment]
Hehe, it will have to wait until monday.  I've not got access to my laptop until then

**Comment by matthid** [comment]
Ok, maybe I'm trying to help by sending a pr to your branch, last time I tried merging directly it looked weird in the pr...

**Comment by matthid** [comment]
(I'm also trying to debug why azure pipelines are red on all prs with this)

**Comment by matthid** [comment]
@BlythMeister not sure if you meant that but I already opened https://github.com/BlythMeister/FAKE/pull/2

**Comment by BlythMeister** [comment]
Merged...I hope

**Comment by matthid** [comment]
@BlythMeister almost... let me try again with https://github.com/BlythMeister/FAKE/pull/3 :)

**Comment by BlythMeister** [comment]
Merged again 😂

**Comment by matthid** [comment]
Lol now the diff apparently fixed itself and is now small again oO did you do things differently this time?

**Comment by BlythMeister** [comment]
Wooooo

**Comment by matthid** [comment]
Will send a pr for all your open prs, go through all of them and start the 5.9 release

**Comment by matthid** [comment]
looks good, thanks!
